es6/time-stamp.js

```diff
@@ -9,16 +9,17 @@ export default (Model, bootOptions = {}) => {
     updatedAt: 'updatedAt',
     required: true,
     validateUpsert: false, // default to turning validation off
+    silenceWarnings: false
   }, bootOptions);
 
   debug('options', options);
 
-  if (!options.validateUpsert && Model.settings.validateUpsert) {
+  if (!options.silenceWarnings && !options.validateUpsert && Model.settings.validateUpsert) {
```

Comment:
This is making the control statement a little too complex.  It also has the effect that the next line `Model.settings.validateUpsert = false;` won't be run if you turn on silenceWarnings.
Perhaps just create a `function warn(options, msg)` and replace the console.warn calls with that.